### ISSUE TITLE
Nothing happens after deploy

### ISSUE THREAD

**Issue by BarryThrill** [aderaaij/wp-vercel-deploy-hooks#1]
Hello!

As a noob with both wordpress and Vercel. It could very well be that I have done anything wrong. Before I start I would like to explain what I did:

1. Created wordpress through XAMPP and I was able to install everything that were needed.
2. I downloaded a template for Wordpress which I edited. When I click on "Show your website" it will show me the webpage I have edited.
3. I went to "addition" and added your deploy hooks through wordpress search.
4. in Vercel. I created a new github with empty folder (Just a readme file) which there after I was able to create hooks through Vercel. etc:

`https://api.vercel.com/v1/integrations/deploy/QmX1255eafaef18294712gZYd8 ` its a fake one :P

5. I navigated to "Deploy" -> "Setting and added the hook with "Activate deploy on post update"
6. I went back to Deploy -> Deploy and clicked on Build site and waited 15 min.

And no changes happend either on Vercel or Github.

Is there anything I am missing here? :) My guess is that the WP needs to automatic update to Github where Vercel reads from it?

Edit:

I do see that the POST gets triggered when clicking on the Build site however it seems that it tries to read from the Github folder which is empty. I assume im missing something in between here?

**Comment by aderaaij** [comment]
Hia, thank you so much for taking the time and reporting this issue in such a detailed manner!

Could you tell me if you get a response from the `POST`? When you click the 'deploy' button, you should get a response like this from the request: 

```js
{"job":{"id":"QC08FLiAu2l59o4LGWYE","state":"PENDING","createdAt":1604934232201}}
```
I wonder if there's a clue in the response here (if there's a response at all). You could also try to use your deploy hook URL in a service like Postman and see if it triggers something and / or gives you some sort of response. The plugin is in no way connected to Github, the connection between your Repository and Vercel is all managed when creating those authorisations. 

Maybe it is indeed because your Github repository is empty and as such, there's nothing to build. I wonder what would happen if you use one of the NextJS examples i.e. https://github.com/vercel/next.js/tree/canary/examples/with-framer-motion. You should be able to deploy it to your own Vercel instance with the blue deploy button in the readme. 

Let me know what happens, thank you!

PS - This plugin could definitely use some better status / error handling, but I haven't exactly figured out how yet 😅

**Comment by BarryThrill** [comment]
> Hia, thank you so much for taking the time and reporting this issue in such a detailed manner!
> 
> Could you tell me if you get a response from the `POST`? When you click the 'deploy' button, you should get a response like this from the request:
> 
> ```js
> {"job":{"id":"QC08FLiAu2l59o4LGWYE","state":"PENDING","createdAt":1604934232201}}
> ```
> 
> I wonder if there's a clue in the response here (if there's a response at all). You could also try to use your deploy hook URL in a service like Postman and see if it triggers something and / or gives you some sort of response. The plugin is in no way connected to Github, the connection between your Repository and Vercel is all managed when creating those authorisations.
> 
> Maybe it is indeed because your Github repository is empty and as such, there's nothing to build. I wonder what would happen if you use one of the NextJS examples i.e. https://github.com/vercel/next.js/tree/canary/examples/with-framer-motion. You should be able to deploy it to your own Vercel instance with the blue deploy button in the readme.
> 
> Let me know what happens, thank you!
> 
> PS - This plugin could definitely use some better status / error handling, but I haven't exactly figured out how yet 😅

Hello! I feel honored to be the first one to post an issue here :D I will try to do all the steps but before that I will start by adding the response I am getting when I click on Build site:

```
{"job":{"id":"FF2OEztuv62pbsj0mfdU","state":"PENDING","createdAt":1604935334795}}

**Comment by BarryThrill** [comment]
Alright after testing, I think it was something that we might think of. What I started to do was to click the Deploy button that you linked me through: [](https://github.com/vercel/next.js/tree/canary/examples/with-framer-motion) - It created a Clone to my github and then got deployed (Its a Motion website: https://test-issue-nrhgbttvw.vercel.app/) - When I do any changes now in Github, I can see through Vercel that it does a triggered with POST and it does what I thought of, it deploys the Github and not the Wordpress. Meaning that everytime I change something in Worpress, it read the Github instead and then deploys the github files instead of the Wordpress.

As I can see there isn't really a connection between Wordpress at all in that case? I guess you need to have your files in the Git and when you do any changes in Wordpress -> Push to Git and then deploy in that case? But I don't believe this is the way it supposed to do?

**Comment by aderaaij** [comment]
Ah I was just responding to you, because indeed I'm thinking you might be under the impression this Plugin does more magic than it actually does!

What this plugin does is: trigger the build and deployment of a static website project on Vercel (like Next.js) website from your WordPress site so you can deploy apps that use your WordPress installation as API / headless Content Management System. 

What it doesn't do, is deploy your WordPress site itself to Vercel. Vercel has no support for databases and such, and is, as far as I gather, exclusively used for front-end code. 

So basically, this plugin offers a way to deploy a static site / React / Vue / whatever app based on your WordPress site. For example: example have one example WordPress installation running on https://wp-headless.arden.nl/ which does not have a theme and is only used as a CMS. On Vercel, I run this repository: https://github.com/aderaaij/wp-headless-client-nextjs which is basically a NextJS app using my WordPress installation as a CMS. Together, they make this example site - https://nextjs-wordpress-ts-example.vercel.app/. 

I hope this clears up any confusion!

**Comment by BarryThrill** [comment]
> Ah I was just responding to you, because indeed I'm thinking you might be under the impression this Plugin does more magic than it actually does!
> 
> What this plugin does is: trigger the build and deployment of a static website project on Vercel (like Next.js) website from your WordPress site so you can deploy apps that use your WordPress installation as API / headless Content Management System.
> 
> What it doesn't do, is deploy your WordPress site itself to Vercel. Vercel has no support for databases and such, and is, as far as I gather, exclusively used for front-end code. 
> 
> So basically, this plugin offers a way to deploy a static site / React / Vue / whatever app based on your WordPress site. For example: example have one example WordPress installation running on https://wp-headless.arden.nl/ which does not have a theme and is only used as a CMS. On Vercel, I run this repository: https://github.com/aderaaij/wp-headless-client-nextjs which is basically a NextJS app using my WordPress installation as a CMS. Together, they make this example site - https://nextjs-wordpress-ts-example.vercel.app/.
> 
> I hope this clears up any confusion!

Ah then here we are! Then I know now what it supposed to do and I had a feeling I was way too noob for this :P But im glad to actually figure this out. 

As I am a rookie with WP itself, I will start dig it more and maybe one day I will understand enough to apply this amazing Deploy :) I really appreciate that you answered me and gave me a really good answer aswell!

I wish you a very good day and stay safe! :D Was a pleasure!

**Comment by aderaaij** [comment]
no worries at all, it seems that you're well on your way to making awesome things, and WordPress is well worth the time looking into, working with. I'm not the biggest expert on WordPress either, but if I can offer any advice, feel free to contact me.

Thank you for your kind words, stay safe as well! 👍
